File: solutions/nim/sum-of-multiples/1/sum_of_multiples.nim
import sequtils

proc sum*(number: int, factors: seq[int]): int =
    let filteredFactors = filter(factors, proc (factor: int): bool = factor != 0)
    let numbers = 1..<number
    for num in numbers:
        if any(filteredFactors, proc (factor: int): bool = num mod factor == 0):
            result += num

File: solutions/r/sum-of-multiples/1/sum-of-multiples.R
sum_of_multiples <- function(factors, limit) {
    multiples_per_factor <- sapply(
        factors,
        function(factor) seq(from = 0, to = limit - 1, by = factor)
    )
    multiples <- unique(unlist(multiples_per_factor))
    sum(multiples)
}
